src/serializer/ResidualHeapSerializer.js

```diff
@@ -827,18 +866,20 @@ export class ResidualHeapSerializer {
     }
 
     invariant(!(val instanceof NativeFunctionValue), "all native function values should be intrinsics");
+    invariant(val instanceof ECMAScriptSourceFunctionValue);
 
     let residualBindings = this.residualFunctionBindings.get(val);
     invariant(residualBindings);
 
-    invariant(val instanceof ECMAScriptSourceFunctionValue);
     let serializedBindings = Object.create(null);
     let instance: FunctionInstance = {
       serializedBindings,
       functionValue: val,
       scopeInstances: new Set(),
     };
 
+    let additionalFuncValueEffects = this.additionalFunctionValuesAndEffects;
+    if (this.currentFunctionBody !== this.mainBody) instance.declarationBodyOverride = this.currentFunctionBody;
```

Comment:
Looks like this would also set `.declarationBodyOverride` for a (nested) residual function that's only reachable from an outer additional function?